src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,426 @@
+        if (!edgeStatus.ok()) {
+            status = edgeStatus.status();
+            break;
+        }
+        auto edgeType = edgeStatus.value();
+        schema_ = ectx()->schemaManager()->getEdgeSchema(spaceId, edgeType);
+        if (schema_ == nullptr) {
+            status = Status::Error("No schema found for '%s'", edgeTypeName_->c_str());
+            break;
+        }
+        edge_.set_edge_type(edgeType);
+
+        status = prepareSet();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareWhere();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareYield();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareNeededProps();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    return status;
+}
+
+
+Status UpdateEdgeExecutor::prepareSet() {
+    auto status = Status::OK();
+    auto items = sentence_->updateList()->items();
+    for (auto& item : items) {
+        // item: name_label ASSIGN expression
+        storage::cpp2::UpdateItem updateItem;
+        updateItem.name = *edgeTypeName_;
+        auto propName = item->field();
+        if (schema_->field(*propName) == nullptr) {
+            status = Status::Error("The schema '%s' does not contain the property '%s'",
+                                   edgeTypeName_->c_str(), propName->c_str());
+            break;
+        }
+        updateItem.field = *propName;
+        item->value()->setContext(expCtx_.get());
+        status = item->value()->prepare();
+        if (!status.ok()) {
+            status = Status::Error("Invalid update item value expression: "
+                                   + item->toString());
+            break;
+        }
+        updateItem.value = Expression::encode(item->value());
+        updateItems_.emplace_back(std::move(updateItem));
+    }
+    if (updateItems_.empty()) {
+        status = Status::Error("There must be some correct update items.");
+    }
+    return status;
+}
+
+
+Status UpdateEdgeExecutor::prepareWhere() {
+    auto *clause = sentence_->whereClause();
+    if (clause != nullptr) {
+        filter_ = clause->filter();
+    }
+    return Status::OK();
+}
+
+
+Status UpdateEdgeExecutor::prepareYield() {
+    auto *clause = sentence_->yieldClause();
+    if (clause != nullptr) {
+        yields_ = clause->columns();
+    }
+    return Status::OK();
+}
+
+
+Status UpdateEdgeExecutor::prepareNeededProps() {
+    auto status = Status::OK();
+    do {
+        if (filter_ != nullptr) {
+            filter_->setContext(expCtx_.get());
+            status = filter_->prepare();
+            if (!status.ok()) {
+                status = Status::Error("Invalid when clause: " + filter_->toString());
+                break;
+            }
+        }
+        if (yields_.empty()) {
+            break;
+        }
+        for (auto *col : yields_) {
```

Comment:
Ditto.  The yield expressions should be calculated on storage service.  We don't need this code